go-controller/pkg/util/multi_network.go

```diff
@@ -0,0 +1,321 @@
+// of the secondary network of the given network name, in the form of <netName>_.
+//
+// Note that for port_group and address_set, it does not allow the '-' character,
+// which will be replaced with ".". Also replace "/" in the nadName with "."
+func GetSecondaryNetworkPrefix(netName string) string {
+	name := strings.ReplaceAll(netName, "-", ".")
+	name = strings.ReplaceAll(name, "/", ".")
+	return name + "_"
+}
+
+func newNetConfInfo(netconf *ovncnitypes.NetConf) (NetConfInfo, error) {
+	if netconf.Name == types.DefaultNetworkName {
+		return &DefaultNetConfInfo{}, nil
+	}
+
+	if netconf.Topology == types.Layer3Topology {
+		return newLayer3NetConfInfo(netconf)
+	}
+	// other topology NAD can be supported later
+	return nil, fmt.Errorf("topology %s not supported", netconf.Topology)
+}
+
+// ParseNADInfo parses config in NAD spec and return a NetAttachDefInfo object for secondary networks
+func ParseNADInfo(netattachdef *nettypes.NetworkAttachmentDefinition) (NetInfo, NetConfInfo, error) {
+	netconf, err := ParseNetConf(netattachdef)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	netconfInfo, err := newNetConfInfo(netconf)
+	if err != nil {
+		return nil, nil, err
+	}
+	return NewNetInfo(netconf), netconfInfo, nil
+}
+
+// ParseNetConf returns NetInfo for the given netconf
+func NewNetInfo(netconf *ovncnitypes.NetConf) NetInfo {
+	// default network netInfo is nil
+	nInfo := DefaultNetworkNetNameInfo
+	if netconf.Name != types.DefaultNetworkName {
+		nInfo = &NetNameInfo{
+			netName:  netconf.Name,
+			nadNames: &sync.Map{},
+		}
+	}
+	return nInfo
+}
+
+// ParseNetConf parses config in NAD spec for secondary networks
+func ParseNetConf(netattachdef *nettypes.NetworkAttachmentDefinition) (*ovncnitypes.NetConf, error) {
+	netconf, err := config.ParseNetConf([]byte(netattachdef.Spec.Config))
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Network Attachment Definition %s/%s: %v", netattachdef.Namespace, netattachdef.Name, err)
+	}
+	// skip non-OVN NAD
+	if netconf.Type != "ovn-k8s-cni-overlay" {
+		return nil, ErrorAttachDefNotOvnManaged
+	}
+
+	if netconf.Name != types.DefaultNetworkName {
+		nadName := GetNADName(netattachdef.Namespace, netattachdef.Name)
+		if netconf.NADName != nadName {
+			return nil, fmt.Errorf("net-attach-def name (%s) is inconsistent with config (%s)", nadName, netconf.NADName)
+		}
+	}
+
+	return netconf, nil
+}
+
+// PodWantsMultiNetwork sees if the given pod needs to plumb over this given network specified by netconf,
+// and return the matching NetworkSelectionElement if any exists.
+//
+// Return value:
+//    bool: if this Pod is on this Network; true or false
+//    *nettypes.NetworkSelectionElement: all NetworkSelectionElement that pod is requested for the specified network
+//    error:  error in case of failure
+func PodWantsMultiNetwork(pod *kapi.Pod, nInfo NetInfo) (bool, *nettypes.NetworkSelectionElement, error) {
```

Comment:
not true, for default network, it returns true with returned network selection being nil.

Reply:
would you point to that return statement ? Is it https://github.com/ovn-org/ovn-kubernetes/blob/98e8dddc513c73264649c6c1d93b51c2f52241c7/go-controller/pkg/util/multi_network.go#L292 ? 

I just saw it returned `true, network`, but didn't check the function that retrieved the network could return a `nil, nil` , my bad. Sorry.

IMO, it could make sense to return an error when `GetK8sPodDefaultNetwork` doesn't find a network instead . Not sure that would complicate matters further elsewhere.

Reply:
by default, even if the default network annoation is not defined, we assume the default network will be plumbed by OVN cni.

Reply:
if the default network is there, why don't we return it ? IMO, it's confusing.

I still think the return values could be coalesced into `*nettypes.NetworkSelectionElement, error`. 

Not going to insist further.

Reply:
no, I meant when Pod does not have the "v1.multus-cni.io/default-network" annotation, it means there is no default network nad associated with this Pod, we'd still return on=true, error=nil but network=nil.

Reply:
Right, I guess returning a dummy network selection element just to simplify the signature would be equally (if not more ...) confusing. Let's leave it as is.

Please resolve this comment.